src/pages/ItemDetail/ItemDetail.js

```diff
@@ -1,8 +1,171 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './ItemDetail.scss';
 
 const ItemDetail = () => {
-  return <div />;
+  const [itemData, setItemData] = useState({});
+  // 아이템 데이터 state구조
+  const [currentTab, setCurrentTab] = useState('상품상세');
+  // 현재 탭 state 구조
+  const [quantity, setQuantity] = useState(1);
+  // 수량 데이터 state구조
+  useEffect(() => {
+    fetch('data/itemData.json')
+      .then(res => res.json())
+      .then(data => setItemData(data));
+  }, []);
+  //목 데이터를 가져와서 아이템 데이터로 설정
+  const {
+    first_category,
+    second_category,
+    title,
+    description,
+    discount,
+    price,
+    thumbnail_images,
+    detail_images,
+  } = itemData;
+  //구조 분해 할당하여 가독성 향상
+
+  const plusPrice = () => {
+    setQuantity(quantity + 1);
+  };
+  const minusPrice = () => {
+    setQuantity(quantity - 1);
+  };
+  //수량 1씩 증가 및 1씩 감소
+  const disabledMinus = quantity < 2;
+  //수량이 2보다 작을때 마이너스 안눌리게 설정
+  const updateCurrentTab = ({ target }) => {
+    const currentTabName = target.innerHTML;
+    setCurrentTab(currentTabName);
+  };
+  //현재 탭에 이름을 부여하도록 설정
+  const discountPrice = price * ((100 - discount) / 100);
+
+  return Object.keys(itemData).length > 0 ? (
+    //아이템 데이터가 객체형식이므로 객체의 키값의 길이가 0보다클때 조건부 렌더링
+    <>
```

Comment:
컴포넌트가 return하는 JSX 자체를 조건부로 렌더링하는 것보다는, 조건에 맞지 않을 때 early return하고 조건이 올바를 때는 올바르게 그려주는 방식으로 구현하는 게 개인적으로는 더 가독성이 좋을 것 같습니다!
정답이 있는 것은 아니니 참고만 해주세요!
```suggestion
  const isData = Object.keys(itemData).length !== 0;
  
  if(!isData) return <>loading...</>
  
  return (
    <>
```